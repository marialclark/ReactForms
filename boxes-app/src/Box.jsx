import './styles/Box.css';

const Box = ({ id, width, height, backgroundColor, removeBox }) => {
  const handleRemove = () => removeBox(id);
  
  return (
    <div className="Box">
      <div className="Box-inner-div"
        style={{
          width: `${width}px`,
          height: `${height}px`,
          backgroundColor: backgroundColor,
        }} 
      >
      </div>
      <button className="Box-remove" onClick={handleRemove}>X</button>
    </div>
  );
}

export default Box;