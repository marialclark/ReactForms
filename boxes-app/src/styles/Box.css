/* Outer box styling */
.Box {
  position: relative;
  display: inline-block;
  box-shadow: 0 2px 4px rgba(0, 0, 0, 0.1);
  margin: 10px;
}

/* Inner box styling */
.Box-innder-div {
  width: 100%;
  height: 100%;
  border-radius: 5px;
}

/* 'X' button for removing the box */
.Box-remove {
  position: absolute;
  top: 5px;
  right: 5px;
  background-color: gray;
  color: white;
  border: none;
  border-color: black;
  border-radius: 50%;
  width: 20px;
  height: 20px;
  cursor: pointer;
  font-size: 14px;
  line-height: 20px;
  text-align: center;
  padding: 0;
}
