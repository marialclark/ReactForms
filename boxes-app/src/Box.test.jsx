import React from 'react';
import { render, screen, fireEvent } from '@testing-library/react';
import Box from './Box';

// Smoke Test
it('renders without crashing', () => {
  render(<Box id="test-id" width={100} height={100} backgroundColor="red" removeBox={() => {}} />);
});

// Snapshot Test
it('matches snapshot', () => {
  const { asFragment } = render(
    <Box id="test-id" width={100} height={100} backgroundColor="red" removeBox={() => {}} />
  );
  expect(asFragment()).toMatchSnapshot();
});

// Critical Business Logic: Removing a Box
it('calls removeBox when the "X" button is clicked', () => {
  const mockRemoveBox = jest.fn();
  render(<Box id="test-id" width={100} height={100} backgroundColor="red" removeBox={mockRemoveBox} />);

  const removeButton = screen.getByText('X');
  fireEvent.click(removeButton);

  expect(mockRemoveBox).toHaveBeenCalledTimes(1);
  expect(mockRemoveBox).toHaveBeenCalledWith('test-id');
});
