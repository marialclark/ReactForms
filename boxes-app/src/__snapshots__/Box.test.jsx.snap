// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="Box"
  >
    <div
      class="Box-inner-div"
      style="width: 100px; height: 100px; background-color: red;"
    />
    <button
      class="Box-remove"
    >
      X
    </button>
  </div>
</DocumentFragment>
`;
